### ISSUE TITLE
[bug]  check_loop_watchdog is not disabled by ATLAS_DISABLE_WATCHDOGS=1

### ISSUE THREAD

**Issue by dribagnac** [Avarok-Cybersecurity/atlas#1135]
### Atlas version / docker tag

git HEAD

### Model

Qwen/Qwen3.6-35B-A3B-FP8 

### Hardware

1x GB10

### Launch command

```bash
/home/daniel/Ai/Atlas/atlas/target/release/spark serve Qwen/Qwen3.6-35B-A3B-FP8 \
    --host 0.0.0.0 \
    --port 8000 \
    --max-seq-len 65536 \
    --kv-high-precision-layers auto \
    --gpu-memory-utilization 0.90 \
    --scheduling-policy slai \
    --enable-prefix-caching \
	--tool-max-tokens 32768 \
	--max-thinking-budget 32768  \
	--tool-call-parser qwen3_coder  \
	--content-loop-watchdog false \
	--content-loop-min-repeats 100  \
	--speculative \
    --num-drafts 2
```

### Expected behaviour

Expected behavior : 
When export ATLAS_DISABLE_WATCHDOGS=1 
This watchdog should NOT be called (function check_loop_watchdog)

### Actual behaviour

Current behavior : 
set variable : export ATLAS_DISABLE_WATCHDOGS=1 

Log
WARN spark loop watchdog fired – repeated line (fuzzy-match) in post-detector content

Process is stopped by Atlas.
I have 5x this text inside my context "#### Résultats"

**Comment by dribagnac** [comment]
Just an idea, add a parameter to set this limit : 0 = no limit, 1, ...

**Comment by dribagnac** [comment]
Currently changed 
crates/spark-server/src/api/stream_guards.rs
    let exact_occurrences = loop_scan_buf.lines().filter(|l| norm(l) == needle).count();
    if exact_occurrences >= 4 {

Changed 4 to 8 for my needs

**Comment by tbraun96** [comment]
**Reopening — this closed automatically on a fix that covers one of four call
sites, which is the opposite of what was agreed.**

#1199 carried `Closes #1135`, and when it landed inside the composed campaign
#1203 (`d67ba440a6`) at 10:25 GitHub closed this. I had said on #1199 that the
issue would be left open on exactly this ground; the auto-close ran anyway.

What #1199 actually fixed: the API-layer **token-loop** watchdog in
`chat_stream/handle_token.rs`, gated on the existing strict parser
`scheduler::parse_disable_watchdogs` — correctly, and without adding a second
implementation of the flag.

What still fires with `AVAROK_DISABLE_WATCHDOGS=1` set:

| still armed | fires at |
|---|---|
| think-skip hard-stop | `scheduler/decode_logits_step.rs:413-421`, MTP twin `emit_step.rs:111-116` — no `disable_watchdogs` check at all |
| fuzzy-repetition | `decode_logits_step.rs:954` → `:979-982` — gated only on the separate `SchedLevers::loop_watchdog` atomic, unlike the content-loop watchdog a few hundred lines away which checks **both** |
| SimHash semantic-loop | `api/chat_stream/handle_token.rs:665` → `:689-706` — gated only on its own `AVAROK_SIMHASH_LOOP` |

So the flag is documented, and logged at boot, as disabling the auto-watchdogs,
and three of them still fire. The remaining work — plus a loose
`AVAROK_SIMHASH_LOOP` parse in the same file and the mirrored parser in
`spark-model` that nothing keeps in sync — is specified with citations in
**#1204**.

This stays open until those three are gated. Nothing for @aniruddh-krovvidi to
do: their change is in and is correct within its stated scope.
